### PR TITLE
Add 2px margin to all widgets but containers

### PR DESCRIPTION
**Pull request by SylvainCorlay**
- Adding missing semantic css classes to widgets
- Give a default margin of 2px to all widgets but containers (hbox, vbox, proxy).

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
@ellisonbg

File: ipywidgets/static/widgets/js/widget_bool.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ define([
              * Called when view is rendered.
              */
             this.$el
-                .addClass('widget-hbox widget-checkbox');
+                .addClass('ipy-widget widget-hbox widget-checkbox');
```
Comment (jdfreder) [inline ipywidgets/static/widgets/js/widget_bool.js]: I'd think `ipywidget` would make more sense, since that's the name of the repo, no?

File: ipywidgets/static/widgets/less/widgets.less [hunk 2]
```diff
@@ -178,37 +180,34 @@
...
 
 .widget-listbox {
     /* Listbox */
-    width         : @widget-width;
-    margin-bottom : 0px;
-    /* Bootstrap does some weird stuff with the margins of select boxes.  The 
-    specifity of the media selector needs to be overwritten using important */
-    margin-left   : 0px !important; 
-    margin-right  : 0px !important;
+    width : @widget-width;
```
Comment (SylvainCorlay) [inline ipywidgets/static/widgets/less/widgets.less]: Your change here was removed in the rebase. I can ad it back I guess but, it should probably be `2px`.

Comment (ellisonbg) [inline ipywidgets/static/widgets/less/widgets.less]: I was thinking that `ipy-widget` makes the most sense as `ipy` is the org
and `widget` is the project. You could imagine `ipy-foo` and `ipy-bar`
eventually from seperate repos.

On Thu, Oct 29, 2015 at 5:29 PM, Sylvain Corlay <notifications@github.com>
wrote:

> In ipywidgets/static/widgets/less/widgets.less
> <https://github.com/ipython/ipywidgets/pull/216#discussion_r43446895>:
>
> >  }
> >
> >  .widget-listbox {
> >      /* Listbox */
> > -    width         : @widget-width;
> > -    margin-bottom : 0px;
> > -    /* Bootstrap does some weird stuff with the margins of select boxes.  The
> > -    specifity of the media selector needs to be overwritten using important */
> > -    margin-left   : 0px !important;
> > -    margin-right  : 0px !important;
> > +    width : @widget-width;
>
> Your change here was removed in the rebase. I can ad it back I guess but,
> it should probably be 2px.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ipython/ipywidgets/pull/216/files#r43446895>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by SylvainCorlay** [comment]
It seems that adding a semantic top-level class breaks the tests.

**Comment by SylvainCorlay** [comment]
This is ready.

**Comment by ellisonbg** [comment]
Great, merging. We can iterate if we find any edge cases.
